pkg/actuators/machine/reconciler.go

```diff
@@ -472,6 +480,17 @@ func (r *Reconciler) requeueIfInstancePending(instance *ec2.Instance) error {
 	return nil
 }
 
+// Check if an instance is terminated so that we can handle it appropriately
+func (r *Reconciler) checkIfInstanceTerminated(instance *ec2.Instance) bool {
+
+	if instance.State != nil && *instance.State.Name == ec2.InstanceStateNameTerminated {
```

Comment:
Use a dereference helper for instance state name, also, check instance is not nil
```suggestion
	if instance != nil && instance.State != nil && aws.StringValue(instance.State.Name) == ec2.InstanceStateNameTerminated {
```